spark/spark-spa/src/main/java/com/thoughtworks/go/spark/spa/AccessTokensController.java

```diff
@@ -1,55 +1,86 @@
-  public AccessTokensController(SPAAuthenticationHelper authenticationHelper, TemplateEngine engine) {
-    this.authenticationHelper = authenticationHelper;
-    this.engine = engine;
-  }
-
-  @Override
-  public String controllerBasePath() {
+    private final SPAAuthenticationHelper authenticationHelper;
+    private final AuthorizationExtensionCacheService authorizationExtensionCacheService;
+    private final SecurityAuthConfigService securityAuthConfigService;
+    private final TemplateEngine engine;
+
+    public AccessTokensController(SPAAuthenticationHelper authenticationHelper,
+                                  AuthorizationExtensionCacheService authorizationExtensionCacheService,
+                                  SecurityAuthConfigService securityAuthConfigService,
+                                  TemplateEngine engine) {
+        this.authenticationHelper = authenticationHelper;
+        this.authorizationExtensionCacheService = authorizationExtensionCacheService;
+        this.securityAuthConfigService = securityAuthConfigService;
+        this.engine = engine;
+    }
+
+    @Override
+    public String controllerBasePath() {
         return Routes.AccessTokens.SPA_BASE;
-  }
-
-  @Override
-  public void setupRoutes() {
-     path(controllerBasePath(), () -> {
-        before("", authenticationHelper::checkUserAnd403);
-        get("", this::index, engine);
-    });
-  }
-  public ModelAndView index(Request request, Response response) {
-      Map<Object, Object> object = new HashMap<Object, Object>() {{
-          put("viewTitle", "Access Tokens");
-      }};
-      return new ModelAndView(object, null);
-  }
+    }
+
+    @Override
+    public void setupRoutes() {
+        path(controllerBasePath(), () -> {
+            before("", authenticationHelper::checkUserAnd403);
+            get("", this::index, engine);
+        });
+    }
+
+    public ModelAndView index(Request request, Response response) {
+        Map<Object, Object> object = new HashMap<Object, Object>() {{
+            put("viewTitle", "Access Tokens");
+            put("meta", getMeta(request));
+        }};
+        return new ModelAndView(object, null);
+    }
+
+    private Map<String, Object> getMeta(Request request) {
+        final AuthenticationToken<?> authenticationToken = SessionUtils.getAuthenticationToken(request.raw());
+        final Map<String, Object> meta = new HashMap<>();
+        meta.put("pluginId", authenticationToken.getPluginId());
+        meta.put("supportsAccessToken", supportsAccessToken(authenticationToken));
+        return meta;
+    }
+
+    private boolean supportsAccessToken(AuthenticationToken<?> authenticationToken) {
+        final SecurityAuthConfig profile = securityAuthConfigService.findProfile(authenticationToken.getAuthConfigId());
+        return authorizationExtensionCacheService.isValidUser(authenticationToken.getPluginId(), authenticationToken.getUser().getUsername(), profile);
```

Comment:
An FYI edge case

A Github user logs into GoCD and has a session. (valid for 30 mins)
During this time, the user is removed from the Github team (by an admin),  When the user lands on the access token SPA, it will say the plugin `github` does not support access token creation, where the reality is:
The plugin supports access token creation but the user is invalid.

Reply:
@GaneshSPatil - User will not get that message as long as we have cached user data, beyond that yes user, will get that message. This implementation is for the current release only. Going forward we will have a specific capability.